### ISSUE TITLE
pint 0.24.3 - Python 3.11.9 error: cannot inherit frozen dataclass from a non-frozen one

### ISSUE THREAD

**Issue by Zippy1999** [hgrecco/pint#2074]
I am seeing the same problem but on python `3.11.9`  using pint 0.24.3

```
<module>
    @dataclass(frozen=True)
../../../../.pyenv/versions/3.11.9/lib/python3.11/dataclasses.py:1222: in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
../../../../.pyenv/versions/3.11.9/lib/python3.11/dataclasses.py:993: in _process_class
    raise TypeError('cannot inherit frozen dataclass from a '
E   TypeError: cannot inherit frozen dataclass from a non-frozen one
```

Everything was working yesterday but after reinstalling the project's virtual env / deps this morning,  I am seeing this error

The file in question throwing the issue is `lib/python3.11/site-packages/pint/delegates/txt_defparser/common.py` on line 23

```
@dataclass(frozen=True)
class DefinitionSyntaxError(errors.DefinitionSyntaxError, fp.ParsingError):
```

**Comment by LorenzoRinaldi** [comment]
same problem here: it works if you downgrade flex parser to 0.3.1

**Comment by Zippy1999** [comment]
You just beat me to it 👍

**Comment by Zippy1999** [comment]
Looking at the code there was a change in master to bump flexparser but fix the affected classes 

https://github.com/hgrecco/pint/commit/d592aff209a0eeae9383042368906dfb5edc5f54#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552

**Comment by stefdoerr** [comment]
Also started getting the error suddenly today which coincides with the release of flexparser.

**Comment by josteinl** [comment]
Need to pin version 0.24.x to flexparser < 0.4.

**Comment by stefdoerr** [comment]
Someone made a conda-forge patch for the current release https://anaconda.org/conda-forge/pint/files

**Comment by jasper-tms** [comment]
For anyone who wants a copy-paste, this downgrade is what has been suggested above, and works perfectly well for me too:

    pip install --upgrade 'flexparser<0.4'

**Comment by hgrecco** [comment]
I apologize for the inconvenience. This change was needed due to a modification in the semantics of frozen dataclass in Python 3.13. It will be fixed with the release today.

**Comment by hgrecco** [comment]
I am closing this issue as is fixed in Pint 0.24.4 Please let me know if you encounter any problem.

**Comment by Zippy1999** [comment]
@hgrecco apologies for the late reply - thanks for looking into this 👍 Keep up the excellent work, it is much appreciated.
